test_conformance/math_brute_force/binary_half.cpp

```diff
@@ -0,0 +1,879 @@
+//
```

Comment:
Most of the newly added files in this commit are based on older versions of the float/double counterparts. There have been many improvements to the original files since this PR was opened, especially regarding kernel creation, use of cl wrapper objects, and removal of `goto`s.  It would be good to base the new `_half.cpp` files on a newer version of the `_float.cpp`/`_double.cpp` files, which would avoid a lot of (bad) code duplication.

(we might wish to wait for #1634 to land before working on this).

Reply:
I would probably start afresh from the latest `main` branch.

There are many things you should still be able to reuse from this PR, such as the plumbing in `main.cpp`, tables with special values and the reference result computations.

Kernel code generation should be much easier now: it should only be a matter of adding a half `ParameterType` to `common.h` and `common.cpp`.

Reply:
I've added adaptation for [#1634](https://github.com/KhronosGroup/OpenCL-CTS/pull/1634) in new PR [#1681](https://github.com/KhronosGroup/OpenCL-CTS/pull/1681).